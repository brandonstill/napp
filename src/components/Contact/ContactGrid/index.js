import _ from 'lodash';
import React from 'react';

import { withStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import Grow from '@material-ui/core/Grow';

import Gravatar from 'react-gravatar';

const styles = {
  card: {
    width: '100%'
  },
  media: {
    objectFit: 'cover',
  },
};

const ContactGrid = props => {

  const classes = props.classes;

  const renderContacts = () => {
    const orderedContacts = _.orderBy(
      props.contacts, 
      [contact => contact[props.sortBy].toLowerCase()], 
      ['asc']
    );

    // return _.map(orderedContacts, contact => {
    //   return (
    //     <li className="list-group-item" key={contact._id}>
    //       {contact.firstName} {contact.lastName} {contact.email}
    //     </li>
    //   );
    // });

    return _.map(orderedContacts, (contact, i) => {
      return (
        <Grid item xs={12} sm={12} md={3} lg={3}>
          <Grow 
            in={true}
            style={{ transformOrigin: '0 0 0' }}
            timeout={i * 200}
          >
            <Card className={classes.card}>
              <Gravatar style={{ width: '100%', height: 'auto' }} email={contact.email} size={500} />
              <CardContent>
                <Typography gutterBottom variant="headline" component="h2">
                {contact.firstName} {contact.lastName}
                </Typography>
                <Typography component="p">
                {contact.email}
                </Typography>
              </CardContent>
            </Card>
          </Grow>
        </Grid>
      );
    });
  }

  return (
    <Grid
      container
      direction="row"
      spacing={16}
    >
      {renderContacts()}
    </Grid>
  )
}

export default withStyles(styles)(ContactGrid);