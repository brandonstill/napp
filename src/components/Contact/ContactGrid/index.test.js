import React from 'react';
import { shallow } from 'enzyme';
import ContactGrid from './';

describe('ContactNav', () => {
  const contacts = [{
    _id: "5ba6ded814f01c02d31cc886",
    email: "email@email.com",
    firstName: "First",
    lastName: "Last"
  }, 
  {
    _id: "5ba6ded814f01c02d31cc887",
    email: "zmail@zmailz.com",
    firstName: "Name",
    lastName: "Nomma"
  }];

  it('renders correctly', () => {
    const wrapper = shallow(<ContactGrid contacts={contacts} sortBy={'lastName'} />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});