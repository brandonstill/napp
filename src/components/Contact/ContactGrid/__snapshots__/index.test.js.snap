// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactNav renders correctly 1`] = `
Array [
  <ContactGrid
    classes={
        Object {
            "card": "ContactGrid-card-1",
            "media": "ContactGrid-media-2",
          }
    }
    contacts={
        Array [
            Object {
              "_id": "5ba6ded814f01c02d31cc886",
              "email": "email@email.com",
              "firstName": "First",
              "lastName": "Last",
            },
            Object {
              "_id": "5ba6ded814f01c02d31cc887",
              "email": "zmail@zmailz.com",
              "firstName": "Name",
              "lastName": "Nomma",
            },
          ]
    }
    sortBy="lastName"
/>,
]
`;
